src/battle_controller_opponent.c

```diff
@@ -1615,6 +1616,10 @@ static void OpponentHandleChooseMove(void)
                 BtlController_EmitTwoReturnValues(BUFFER_B, 10, (chosenMoveId) | (gActiveBattler << 8));
             else if (gBattleTypeFlags & BATTLE_TYPE_DOUBLE)
             {
+                do {
+                    target = GetBattlerAtPosition(Random() & 2);
+                } while (!CanTargetBattler(gActiveBattler, target, move));
+                
```

Comment:
Question for future proofing: if all 3 targets in double battles couldn't be targeted, wouldn't this make an infinite loop?

Reply:
Actually looking at this again, Random() & 2 is only ever 0 or 2, so wild Mon will never target their partner